system/lib/compiler-rt/lib/sanitizer_common/sanitizer_emscripten.cpp

```diff
@@ -128,6 +136,92 @@ u64 MonotonicNanoTime() {
 
 void GetMemoryProfile(fill_profile_f cb, uptr *stats) {}
 
+int internal_madvise(uptr addr, uptr length, int advice) {
+  return 0; // madvise is currently ignored
+}
+
+uptr internal_close(fd_t fd) {
+  return __wasi_fd_close(fd);
+}
+
+uptr internal_open(const char *filename, int flags) {
+  return open(filename, flags);
+}
+
+uptr internal_open(const char *filename, int flags, u32 mode) {
+  return open(filename, flags, mode);
+}
+
+uptr internal_read(fd_t fd, void *buf, uptr count) {
+  __wasi_iovec_t iov = { (uint8_t*)buf, count };
+  size_t num;
+  if (__wasi_syscall_ret(__wasi_fd_read(fd, &iov, 1, &num))) {
+    return -1;
+  }
+  return num;
+}
+
+uptr internal_write(fd_t fd, const void *buf, uptr count) {
+  __wasi_ciovec_t iov = { (const uint8_t*)buf, count };
+  size_t num;
+  if (__wasi_syscall_ret(__wasi_fd_write(fd, &iov, 1, &num))) {
+    return -1;
+  }
+  return num;
+}
+
+uptr internal_stat(const char *path, void *buf) {
+  return stat(path, (struct stat *)buf);
+}
+
+uptr internal_fstat(fd_t fd, void *buf) {
+  return fstat(fd, (struct stat *)buf);
+}
+
+uptr internal_filesize(fd_t fd) {
+  struct stat st;
+  if (internal_fstat(fd, &st))
+    return -1;
+  return (uptr)st.st_size;
+}
+
+uptr internal_dup(int oldfd) {
+  return dup(oldfd);
+}
+
+uptr internal_getpid() {
+  return 42;
+}
+
+uptr internal_sched_yield() {
+  return sched_yield();
+}
+
+void internal_sigfillset(__sanitizer_sigset_t *set) {
+  sigfillset(set);
+}
+
+uptr internal_sigprocmask(int how, __sanitizer_sigset_t *set,
+                          __sanitizer_sigset_t *oldset) {
+  return sigprocmask(how, set, oldset);
+}
+
+void internal_usleep(u64 useconds) {
+  usleep(useconds);
+}
+
+void internal__exit(int exitcode) {
+  __wasi_proc_exit(exitcode);
+}
+
+tid_t GetTid() {
+  return gettid();
```

Comment:
Note: this was previously using `pthread_self()`:
https://github.com/emscripten-core/emscripten/blob/b0c461c10482e12f88e192e04b3640acb1577f64/system/lib/compiler-rt/lib/sanitizer_common/sanitizer_linux.cpp#L683-L684

After commit 77859cd7e125efe23689505f4c181df81738e46d, we can safely use `gettid()` here.